### PR TITLE
Replace text annotations with explicit file/line for debug info

### PR DESCRIPTION
**Pull request by dschuff**
Rather than storing debug info as text annotations, store explicit file and line information. This will make it easier to experiment with outputting other serializations or representations (e.g. source maps), and will allow outputting debug info for binaries as well.

### CODE REVIEW

File: src/wasm.h [hunk 1]
```diff
@@ -511,7 +511,8 @@ class Function {
   std::map<Name, Index> localIndices;
 
   // node annotations, printed alongside the node in the text format
```
Comment (kripken) [inline src/wasm.h]: stale comment

File: src/wasm.h [hunk 2]
```diff
@@ -511,7 +511,8 @@ class Function {
...
-  std::unordered_map<Expression*, std::string> annotations;
+  struct DebugLocation { uint32_t fileIndex, lineNumber; };
```
Comment (kripken) [inline src/wasm.h]: contents of struct should go on a new line, for consistency with the others

**Comment by dschuff** [comment]
No, I'm not sure we'll never want other annotations. But I do know we want a more explicit representation of debug info, and that we don't *currently* want other annotations. So if we want more in the future we can always just add back the annotations mechanism.

**Comment by kripken** [comment]
Yeah, fair enough. lgtm.

**Pull request merged**